app/models/presenter/auction.rb

```diff
@@ -94,8 +94,46 @@ def html_summary
       markdown.render(summary)
     end
 
+    def status
+      if available?
+        'Open'
+      else
+        'Closed'
+      end
+    end
+
+
+    delegate :label_class, :label, :tag_data_value_status, :tag_data_label_2, :tag_data_value_2,
+      to: :status_presenter
+
+    def human_start_time
+      if start_datetime < Time.now
+        # this method comes from the included date helpers
+        "#{distance_of_time_in_words(Time.now, start_datetime)} ago"
+      else
+        "in #{distance_of_time_in_words(Time.now, start_datetime)}"
+      end
+    end
+
     private
 
+    def status_presenter_class
+      status_name = if expiring?
+                      'Expiring'
+                    elsif over?
+                      'Over'
+                    elsif future?
+                      'Future'
+                    else
+                      'Open'
+                    end
+      "::Presenter::AuctionStatus::#{status_name}".constantize
```

Comment:
This is an interesting technique for polymorphism, but I noticed this immediately. So, if I may play Devil's Advocate here, what is the specific advantage of having distinct Ruby classes for these fields that we instantiate like this vs., say,  an equivalently structured hash for looking up such information? I guess the goal is that you can test all these classes with their own unit tests?